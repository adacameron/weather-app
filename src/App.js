import React from "react";
import "./App.css";

function App() {
  return (
    <div className="App">
      <h1>Weather Station</h1>
    </div>
  );
}

export default App;
