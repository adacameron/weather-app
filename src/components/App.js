import React, { useEffect, useState } from "react";
import "../styles/App.css";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";
import ForecastDetails from "./ForecastDetails";
import getForecast from "../requests/getForecast";
import SearchForm from "./SearchForm";
import Header from "./Header";

const App = () => {
  const [location, setLocation] = useState({ city: "", country: "" });

  const [forecasts, setForecasts] = useState([]);

  const [selectedDate, setSelectedDate] = useState(0);

  const [searchText, setSearchText] = useState("");

  const [errorMessage, setErrorMessage] = useState("");

  const selectedForecast = forecasts.find(
    (forecast) => forecast.date === selectedDate
  );

  const handleForecastSelect = (date) => {
    setSelectedDate(date);
  };

  const handleCitySearch = () => {
    getForecast(
      searchText,
      setSelectedDate,
      setForecasts,
      setLocation,
      setErrorMessage
    );
  };

  useEffect(() => {
    getForecast(
      searchText,
      setSelectedDate,
      setForecasts,
      setLocation,
      setErrorMessage
    );
  }, []);

  return (
    <div className="weather-app">
      <Header />
      <LocationDetails
        city={location.city}
        country={location.country}
        errorMessage={errorMessage}
      />
      <SearchForm
        searchText={searchText}
        setSearchText={setSearchText}
        onSubmit={handleCitySearch}
      />
      {!errorMessage && (
        <>
          <ForecastSummaries
            forecasts={forecasts}
            onForecastSelect={handleForecastSelect}
          />
          {selectedForecast && <ForecastDetails forecast={selectedForecast} />}
        </>
      )}
    </div>
  );
};

export default App;
