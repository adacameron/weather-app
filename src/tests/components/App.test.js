import { render } from "@testing-library/react";
import React from "react";
import App from "../../components/App";

describe("App", () => {
  it("renders components correctly", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
