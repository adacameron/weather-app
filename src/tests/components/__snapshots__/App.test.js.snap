// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders components correctly 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <h1>
      Weather Station
    </h1>
    <h1
      class="location-details"
      style="border: 2px solid red;"
    >
      , 
    </h1>
    <div
      class="search-form"
    >
      <input
        type="text"
        value=""
      />
      <button
        class="search-form__button"
        type="submit"
      >
        Search
      </button>
    </div>
    <div
      class="forecast-summaries"
      data-testid="forecast-summaries"
    />
  </div>
</DocumentFragment>
`;
